[Terraform] Updates to align with PCX content strategy (#2238)
cloudflare/cloudflare-docs: products/terraform/src/content/tutorial/use-load-balancing.md

---
title: 5 – Improve performance
order: 5
pcx-content-type: tutorial
---

# Improve performance and reliability

Now that rate limiting is set up, your login page is now protected against credential brute for attacks.

> In this tutorial, you will add a second origin for some basic round robining, and then use the [Cloudflare Load Balancing](https://www.cloudflare.com/load-balancing/) product to fail traffic over as needed. You will also enhance your load balancing configuration through the use of "geo steering" to serve results from an origin server that is geographically closest to your end users.

Reviewer: ```suggestion
In this tutorial, you will add a second origin for some basic round robining, and then use the [Cloudflare Load Balancing](https://developers.cloudflare.com/load-balancing/) product to fail traffic over as needed. You will also enhance your load balancing configuration through the use of "geo steering" to serve results from an origin server that is geographically closest to your end users.
```

Hello, this is 203.0.113.10!
```

As noted above, there is no discernible pattern for which origin receives the request. When Cloudflare connects to an origin with multiple DNS records, one of the IP addresses is selected at random. If both IPs are in the same data center and sessions can be shared (i.e., it doesn't matter if the same user hops between origin servers), this may work fine. However, for anything more complicated, such as origins in different geographies or active health checks, you will want to use Cloudflare's Load Balancing product.

## 4. Switch to using Cloudflare's Load Balancing product

<Aside>

Before proceeding, ensure Load Balancing is enabled for your account. If you are on an Enterprise plan, contact your Customer Success Manager to enable Load Balancing' otherwise, you can subscribe to Load Balancing within the Cloudflare Dashboard.

</Aside>

> As described in the [Load Balancing tutorial](https://support.cloudflare.com/hc/en-us/articles/115000081911-Tutorial-How-to-Set-Up-Load-Balancing-Intelligent-Failover-on-Cloudflare), you will need to complete three tasks:

Reviewer: ```suggestion
As described in the [Load Balancing tutorial](https://developers.cloudflare.com/load-balancing/get-started), you will need to complete three tasks:
```

1. Create a monitor to run health checks against your origin servers.
2. Create a pool of one or more origin servers that will receive load balanced traffic.

> 3. Create a load balancer with an external hostname, e.g., www.example.com, and one or more pools.

Reviewer: ```suggestion
3. Create a load balancer with an external hostname, e.g., `www.example.com`, and one or more pools.
```

### i. Define and create the health check ("monitor")

> To monitor the origins, create a basic health check that makes a GET request to each origin on the URL https://www.example.com. If the origin returns the `200/OK` status code within five seconds, it is considered healthy. If it fails to do so three times in a row, it is considered unhealthy. This health check will be run once per minute from several regions and send an email notification to you@example.com if any failures are detected.

Reviewer: ```suggestion
To monitor the origins, create a basic health check that makes a GET request to each origin on the URL `https://www.example.com`. If the origin returns the `200/OK` status code within five seconds, it is considered healthy. If it fails to do so three times in a row, it is considered unhealthy. This health check will be run once per minute from several regions and send an email notification to you@example.com if any failures are detected.
```

### iii. Define and create the load balancer

> When you create a load balancer (LB), it will [replace any existing DNS records with the same name](https://support.cloudflare.com/hc/en-us/articles/115004954407-How-Does-a-Load-Balancer-Interact-with-Existing-DNS-Records-). For example, if you create the "www.example.com" LB below, it will supersede the two www DNS records that you previously defined. One benefit of leaving the DNS records in place is that if you temporarily disable load balancing, connections to this hostname are still possible as shown in step #2 above.

Reviewer: ```suggestion
When you create a load balancer (LB), it will [replace any existing DNS records with the same name](https://developers.cloudflare.com/load-balancing/reference/dns-records). For example, if you create the `www.example.com` LB below, it will supersede the two www DNS records that you previously defined. One benefit of leaving the DNS records in place is that if you temporarily disable load balancing, connections to this hostname are still possible as shown in step #2 above.
```
